lib/provider.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:flutter/widgets.dart';
+
+/// ChangeNotifier is extending [Listenable] so by using that subclass this can
+/// also be reused for other [Listenable] classes.
+class Provider<T extends Listenable> extends InheritedNotifier<T> {
+  const Provider({
+    super.key,
+    required super.child,
+    required super.notifier,
+  });
+
+  static T of<T extends Listenable>(BuildContext context) {
+    final provider = context.dependOnInheritedWidgetOfExactType<Provider<T>>();
+
+    if (provider == null) {
+      throw Exception("No Provider found in context");
+    }
+
+    final notifier = provider.notifier;
+
+    if (notifier == null) {
+      throw Exception("No notifier found in Provider");
+    }
+
+    return notifier;
+  }
```

Comment:
I wonder if we should actually have another method of just "reading" the provider, maybe that's overkill in this example though.

Reply:
That would be cool, but I think it is a bit overkill for this. Als this matches the article we wrote so if users are coming from that they will recognize it.